tests/fixtures.py

```diff
@@ -0,0 +1,231 @@
+# limitations under the License.
+"""Common fixtures."""
+
+import collections
+import itertools
+import os
+import shutil
+import tempfile
+from pathlib import Path
+
+import pytest
+import tftest
+import yaml
+
+PlanSummary = collections.namedtuple('PlanSummary', 'values counts outputs')
+
+
+def plan_summary(module_path, basedir, tf_var_files=None, **tf_vars):
+  """
+  Run a Terraform plan on the module located at `module_path`.
+
+  - module_path: terraform root module to run. Can be an absolute
+    path or relative to the root of the repository
+
+  - basedir: directory root to use for relative paths in
+    tf_var_files.
+
+  - tf_var_files: set of terraform variable files (tfvars) to pass
+    in to terraform
+
+  Returns a PlanSummary object containing 3 attributes:
+  - values: dictionary where the keys are terraform plan addresses
+    and values are the JSON representation (converted to python
+    types) of the attribute values of the resource.
+
+  - counts: dictionary where the keys are the terraform resource
+    types and the values are the number of times that type appears
+    in the plan
+
+  - outputs: dictionary of the modules outputs that can be
+    determined at plan type.
+
+  Consult [1] for mode details on the structure of values and outputs
+
+  [1] https://developer.hashicorp.com/terraform/internals/json-format
+  """
+  module_path = Path(__file__).parents[1] / module_path
+
+  # FIXME: find a way to prevent the temp dir if TFTEST_COPY is not
+  # in the environment
+  with tempfile.TemporaryDirectory(dir=module_path.parent) as tmp_path:
+    # if TFTEST_COPY is set, copy the fixture to a temporary
+    # directory before running the plan. This is needed if you want
+    # to run multiple tests for the same module in parallel
+    if os.environ.get('TFTEST_COPY'):
+      test_path = Path(tmp_path)
+      shutil.copytree(module_path, test_path, dirs_exist_ok=True)
+
+      # if we're copying the module, we might as well remove any files
+      # and directories from the test directory that are automatically
+      # read by terraform. Useful to avoid surprises if, for example,
+      # you have an active fast deployment with links to configs)
+      autopaths = itertools.chain(
+          test_path.glob('*.auto.tfvars'),
+          test_path.glob('*.auto.tfvars.json'),
+          test_path.glob('terraform.tfstate*'),
+          test_path.glob('terraform.tfvars'),
+          test_path.glob('.terraform'),
+          # any symlinks?
+      )
+      for p in autopaths:
+        if p.is_dir():
+          shutil.rmtree(p)
+        else:
+          p.unlink()
+    else:
+      test_path = module_path
+
+    # prepare tftest and run plan
+    binary = os.environ.get('TERRAFORM', 'terraform')
+    tf = tftest.TerraformTest(test_path, binary=binary)
+    tf.setup(upgrade=True)
+    tf_var_files = [(basedir / x).resolve() for x in tf_var_files or []]
+    plan = tf.plan(output=True, refresh=True, tf_var_file=tf_var_files,
```

Comment:
`refresh=True` by default in tftest

Reply:
wdym? Remove it? It was like that in our existing fixture so I kept it.

Reply:
yep we can remove

Reply:
I'd keep it. It doesn't hurt and it would avoid running tests with old provider versions if you happen to have done an init within the module's directory.